pkgs/development/compilers/ghc/7.10.3.nix

```diff
@@ -16,7 +16,7 @@
 
 , # If enabled, GHC will be built with the GPL-free but slower integer-simple
   # library instead of the faster but GPLed integer-gmp library.
-  enableIntegerSimple ? !(gmp.meta.available or false), gmp
+  enableIntegerSimple ? false, gmp ? null
```

Comment:
IMO ```gmp ? null``` is confusing to new users. gmp will never be null even if gmp is unsupported.